cli/command/stack/list.go

```diff
@@ -69,7 +82,33 @@ func runList(cmd *cobra.Command, dockerCli command.Cli, opts options.List) error
 		Format: formatter.Format(format),
 	}
 	sort.Slice(stacks, func(i, j int) bool {
-		return sortorder.NaturalLess(stacks[i].Name, stacks[j].Name)
+		return sortorder.NaturalLess(stacks[i].Name, stacks[j].Name) ||
+			!sortorder.NaturalLess(stacks[j].Name, stacks[i].Name) &&
+				sortorder.NaturalLess(stacks[j].Namespace, stacks[i].Namespace)
 	})
 	return formatter.StackWrite(stackCtx, stacks)
 }
+
+func getStacks(dockerCli command.Cli, opts options.List, kopts kubernetes.Options) ([]*formatter.Stack, error) {
+	kli, err := kubernetes.WrapCli(dockerCli, kopts)
+	if err != nil {
+		return nil, err
+	}
+	ss, err := kubernetes.GetStacks(kli, opts)
+	if err != nil {
+		return nil, err
+	}
+	return ss, nil
+}
+
+func getNamespaces(cmd *cobra.Command) (map[string]struct{}, error) {
```

Comment:
This could just take (`flags *flag.FlagSet`, or even `namespaces []string`, then call it with `getNamespaces(cmd.Flags())`, but see my other comments

Reply:
This has actually been changed to receive a ```flags *flag.FlagSet``` in a follow-up PR, but I might as well do it there.

Reply:
Hmm sorry I meant ```[]string``` and passing it ```opts.Namespaces```…